File: leetcode_solutions/0027-remove-element/solution.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        p_write = 0
        p_read = 0

        while p_read < len(nums):
            if (nums[p_read] != val):
                nums[p_write] = nums[p_read]
                p_write += 1
            p_read += 1
        return p_write 

File: leetcode_solutions/0027-remove-element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int p_write = 0;
        int p_read = 0;
        while(p_read < nums.length){
            if (nums[p_read] != val){
                nums[p_write++] = nums[p_read];
            }
            p_read++;
        }
        return p_write;
    }
}
